src/aggregator/aggregator/elem_base.go

```diff
@@ -216,6 +216,26 @@ type consumeState struct {
 	resendEnabled bool
 }
 
+// Reset resets the consume state for reuse.
+func (c *consumeState) Reset() {
+	var (
+		annotation []byte
+		values     []float64
+	)
+
+	if c.annotation != nil {
+		annotation = c.annotation[:0]
```

Comment:
am I missing some gc trick here? why can't you just do 

```
c.annotation = c.anntation[:0]
```

Reply:
whoops, I assumed doing:
```
	var p []int
	p = p[:0]
```
will panic